test/widgets/autocomplete_test.dart

```diff
@@ -147,33 +153,54 @@ void main() {
-      checkUserShown(user2, store, expected: true);
-      checkUserShown(user3, store, expected: true);
+      checkUserShown(user1, expected: false);
+      checkUserShown(user2, expected: true);
+      checkUserShown(user3, expected: true);
 
       // Finishing autocomplete updates compose box; causes options to disappear
       await tester.tap(find.text('User Three'));
       await tester.pump();
       check(tester.widget<TextField>(composeInputFinder).controller!.text)
         .contains(mention(user3, users: store.users));
-      checkUserShown(user1, store, expected: false);
-      checkUserShown(user2, store, expected: false);
-      checkUserShown(user3, store, expected: false);
+      checkUserShown(user1, expected: false);
+      checkUserShown(user2, expected: false);
+      checkUserShown(user3, expected: false);
 
       // Then a new autocomplete intent brings up options again
       // TODO(#226): Remove this extra edit when this bug is fixed.
       await tester.enterText(composeInputFinder, 'hello @user tw');
       await tester.enterText(composeInputFinder, 'hello @user two');
       await tester.pumpAndSettle(); // async computation; options appear
-      checkUserShown(user2, store, expected: true);
+      checkUserShown(user2, expected: true);
 
       // Removing autocomplete intent causes options to disappear
       // TODO(#226): Remove one of these edits when this bug is fixed.
       await tester.enterText(composeInputFinder, '');
       await tester.enterText(composeInputFinder, ' ');
-      checkUserShown(user1, store, expected: false);
-      checkUserShown(user2, store, expected: false);
-      checkUserShown(user3, store, expected: false);
+      checkUserShown(user1, expected: false);
+      checkUserShown(user2, expected: false);
+      checkUserShown(user3, expected: false);
+
+      debugNetworkImageHttpClientProvider = null;
+    });
+
+    testWidgets('test delivery email visibility', (tester) async {
+      final user2 = eg.user(userId: 2, fullName: 'User Two', avatarUrl: 'user2.png',);
+      final user3 = eg.user(userId: 3, fullName: 'User Three', avatarUrl: 'user3.png', deliveryEmail: 'email3@email.com');
+      final composeInputFinder = await setupToComposeInput(tester, users: [user2, user3]);
+
+      TypingNotifier.debugEnable = false;
+      addTearDown(TypingNotifier.debugReset);
+
+      // Options are filtered correctly for query
+      // TODO(#226): Remove this extra edit when this bug is fixed.
+      await tester.enterText(composeInputFinder, 'hello @user ');
+      await tester.enterText(composeInputFinder, 'hello @user t');
+      await tester.pumpAndSettle(); // async computation; options appear
+
+      // "User Two"'s delivery email is not visible and "User Three"'s delivery email is visible
+      checkUserShown(user2, expected: true, deliveryEmailExpected: false);
+      checkUserShown(user3, expected: true, deliveryEmailExpected: true);
```

Comment:
Let's have two separate test cases for this (i.e. two `testWidgets` calls), as I said above:
https://github.com/zulip/zulip-flutter/pull/995#pullrequestreview-2437163329

That's useful because it means if some change someone's working on causes these tests to fail in the future, they'll see specifically whether it was one case, the other, or both of them that failed.

It's also useful because it lets each of the test cases be simpler, which makes them easier to read, understand, and update when needed.

Reply:
Okay, understood